### PR TITLE
#160577487 List all users

### PR DESCRIPTION
**Pull request by koechkevin**
#### What does this PR do?
- Have an authorised user view profiles of other users.
- creates a test folder for profiles' app and put all test files for the application

#### Description of Task to be completed?
- Have the following endpoints working
```
GET /api/profiles/
```
#### How should this be manually tested?
- After cloning [this repo](https://github.com/andela/ah-shakas), cd into the directory, checkout to branch ft-list-users-160577487 and run the application,
```git checkout ft-list-users-160577487```
```pipenv install```
```pipenv shell```
```pipenv install```
```
python manage.py migrate
```
```
python manage.py runserver
```
- using postman or any API client, test the endpoint with header:
_key_: Authorization, _value_:Token from  ```POST: /api/users/login/``` with payload in the format 
```
{
	"user": {
		"email": "testinguser@test.com",
		"username": "testinguser",
		"password": "testinguser123"
	}
}
```
- Note that you have to be registered with your account activated. The endpoint for registeration is 
```POST: /api/users/``` with payload in the format 
```
{
	"user": {
		"email": "testinguser@test.com",
		"username": "testinguser",
		"password": "testinguser123"
	}
}
```
#### Any background context you want to provide?
- Tests for this feature is found in authors/apps/profiles/test/test_view_all_profiles.py
- The implementation of this feature was largely dependant on 
https://www.pivotaltracker.com/story/show/160577479
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/160577487
#### Screenshots (if appropriate)
screen shot 2018-10-12 at 16 03 00

### CODE REVIEW

File: authors/apps/profiles/renderers.py [hunk 1]
```diff
@@ -22,4 +22,12 @@ def render(self, data, media_type=None, renderer_context=None):
         return json.dumps({
             'profile': data
         })
-        
+
```
Comment (lukorito) [inline authors/apps/profiles/renderers.py:25]: You should add a single line here: PEP8 guidelines

File: authors/apps/profiles/serializers.py [hunk 2]
```diff
@@ -36,7 +36,19 @@ def update(self, instance, validated_data):
             instance.user.save()
         return instance
 
-   
+
     class Meta:
         model = Profile
         fields = ('username', 'bio', 'image_url', 'following', 'created_at', 'updated_at')
+
```
Comment (lukorito) [inline authors/apps/profiles/serializers.py:43]: ditto

File: authors/apps/profiles/test/test_view_all_profiles.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
+from django.contrib.auth import get_user_model
+from rest_framework.test import APITestCase
+from rest_framework.authtoken.models import Token
+from rest_framework.reverse import reverse
+from rest_framework import status
+import json
+from ..models import Profile
+from ...authentication.models import User
+
```
Comment (lukorito) [inline authors/apps/profiles/test/test_view_all_profiles.py:9]: ditto

File: authors/apps/profiles/views.py [hunk 4]
```diff
@@ -42,3 +42,23 @@ def put(self, request, username, format=None):
                     },
                     status=status.HTTP_404_NOT_FOUND
                 )
+
```
Comment (lukorito) [inline authors/apps/profiles/views.py:57]: ditto

**Review by lukorito — CHANGES_REQUESTED** [review]

File: authors/apps/profiles/renderers.py (hunk 1 above)
Comment (koechkevin) [reply authors/apps/profiles/renderers.py:25]: noted

**Review by michael-basweti — COMMENTED** [review]
In your PR, you can add a link on where you are cloning from so that someone can click and be redirected to the repo

**Review by codingedward — COMMENTED** [review]
I have gone through your implementation and I believe this is ready for a merge. Please rebase to current develop.

Also, let's fix the dropping coverage 😃

**Review by michael-basweti — CHANGES_REQUESTED** [review]
It would be awesome if in your PR you also give the payload and endpoint for signup and also payload for login. A user cannot sign in without having to signup first

File: authors/apps/profiles/views.py [hunk 5]
```diff
@@ -5,15 +5,17 @@
 from rest_framework import status, generics
 from rest_framework.response import Response
 from rest_framework.pagination import LimitOffsetPagination
-from rest_framework.generics import CreateAPIView
+from rest_framework.generics import CreateAPIView, ListAPIView
 from rest_framework import status
+from django.http import Http404
```
Comment (lukorito) [inline authors/apps/profiles/views.py]: This line is imported but not being used

File: authors/apps/profiles/views.py [hunk 6]
```diff
@@ -5,15 +5,17 @@
...
 from authors.apps.authentication.models import User
 from .models import Profile
-from .renderers import ProfileJSONRenderer
-from .serializers import ProfileSerializer
+from .renderers import ProfileJSONRenderer, ProfilesJSONRenderer
+from .serializers import ProfileSerializer, ProfileListSerializer
 from authors.apps.authentication.serializers import UserSerializer
 
+
 # Create your views here.
```
Comment (lukorito) [inline authors/apps/profiles/views.py]: Remove this

**Review by lukorito — CHANGES_REQUESTED** [review]

**Comment by lukorito** [comment]
@koechkevin Great work on the code. However, there are some unused imports you are adding. Check on that

**Comment by caveinn** [comment]
@koechkevin I think you can improve your coverage by testing a user trying to follow a user that does not exist.  Also I think for the PR the part on how to test this manually can be made more descriptive.

**Review by michael-basweti — APPROVED** [review]

**Review by lukorito — APPROVED** [review]

**Pull request merged**